src/lib/metrics/Metrics.ts

```diff
@@ -229,12 +164,74 @@ export class Metrics {
     this.pushEvent(event);
   };
 
+  _processQueue = async () => {
+    this.isProcessing = true;
+
+    if (this.queue.length === 0) {
+      if (this.debug) {
+        // eslint-disable-next-line no-console
+        console.log("Metrics: queue is empty");
+      }
+      return;
+    }
+
+    // Avoid infinite queue growth
+    if (this.queue.length > MAX_QUEUE_LENGTH) {
+      this.queue = this.queue.slice(MAX_BATCH_LENGTH - 1);
+      if (this.debug) {
+        // eslint-disable-next-line no-console
+        console.log("Metrics: Slice queue");
+      }
+    }
+
+    let items;
+    if (this.queue.length > MAX_BATCH_LENGTH) {
+      items = this.queue.slice(0, MAX_BATCH_LENGTH);
+      this.queue = this.queue.slice(MAX_BATCH_LENGTH - 1);
```

Comment:
no need to check `this.queue.length > MAX_BATCH_LENGTH`, just always do this logic

Reply:
not fixed